src/main/java/org/prebid/server/bidder/impactify/ImpactifyBidder.java

```diff
@@ -0,0 +1,203 @@
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.request.User;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.currency.CurrencyConversionService;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.impactify.ExtImpImpactify;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ImpactifyBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpImpactify>> IMPACTIFY_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpImpactify>>() {
+            };
+    private static final String X_OPENRTB_VERSION = "2.5";
+    private static final String BIDDER_CURRENCY = "USD";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    public ImpactifyBidder(String endpointUrl, JacksonMapper mapper, CurrencyConversionService conversionService) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+        this.currencyConversionService = Objects.requireNonNull(conversionService);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> updatedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            BigDecimal bidFloor = imp.getBidfloor();
+            try {
+                if (shouldConvertBidFloor(bidFloor, imp.getBidfloorcur())) {
+                    bidFloor = convertBidFloorCurrency(bidFloor, request, imp.getId(), imp.getBidfloorcur());
+                }
+                updatedImps.add(updateImp(imp, bidFloor));
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
```

Comment:
What if
```java
for (Imp imp : request.getImp()) {
            try {
                final BigDecimal resolvedBidFloor = resolveBidFloor(request, imp, imp.getBidfloor());
                updatedImps.add(updateImp(imp, resolvedBidFloor));
            } catch (PreBidException e) {
                return Result.withError(BidderError.badInput(e.getMessage()));
            }
        }
        
....

private BigDecimal resolveBidFloor(BidRequest request, Imp imp, BigDecimal bidFloor) {
        return shouldConvertBidFloor(bidFloor, imp.getBidfloorcur())
                ? convertBidFloorCurrency(bidFloor, request, imp.getId(), imp.getBidfloorcur())
                : imp.getBidfloor();
    }
```

Reply:
Yea, that is better, obviously! Thank you!